kube: IsReachable panics on second call when client creation fails (typed-nil cached in getKubeClient)

### What happened?

I use Helm as a Go package (`helm.sh/helm/v4` v4.2.0), not the CLI. `kube.(*Client).IsReachable()` panics with a nil pointer dereference when it is called a second time on a `Client` whose Kubernetes client could not be created on the first call.

The root cause is in `getKubeClient`:

```go
func (c *Client) getKubeClient() (kubernetes.Interface, error) {
	var err error
	if c.kubeClient == nil {
		c.kubeClient, err = c.Factory.KubernetesClientSet()
	}
	return c.kubeClient, err
}
```

`KubernetesClientSet()` returns `(*kubernetes.Clientset, error)`. When `ToRESTConfig()` fails (for example, the selected context does not exist, or the config is empty), it returns `(nil, err)`. The `nil` concrete pointer is then stored into the `c.kubeClient` field, which is an interface (`kubernetes.Interface`). A typed nil stored in an interface makes the interface itself non-nil.

Because of this:

1. First call to `getKubeClient`: `c.kubeClient` is nil, so it calls the factory. The factory returns `(nil, err)`. The field now holds a typed nil, and the error is returned. `IsReachable` sees the error and returns a clean "kubernetes cluster unreachable" error. This part is fine.
2. Second call to `getKubeClient`: `c.kubeClient == nil` is now false (the interface is non-nil), so the factory is not called again. It returns the typed-nil client and `err == nil`.
3. Back in `IsReachable`, both error checks pass because the error is nil. The code then runs `client.Discovery().ServerVersion()`. `Discovery()` on the nil `*Clientset` returns nil, and the call to `ServerVersion()` panics.

This is easy to hit through the action package, because more than one action checks reachability on the same `Configuration.KubeClient`. For example, a common install flow first runs `history` (which calls `IsReachable`) and then `install` (which also calls `IsReachable`). The first check returns the error, but the second check panics instead of returning the same error.

This is not a caller-side problem. I only use the Helm Go API. Two calls to `IsReachable` on one `kube.Client`, or one `history` action followed by one `install` action on a single `action.Configuration`, are enough to trigger it. Reusing one `Configuration` across actions is the intended use of the API, so this is normal usage, not misuse.

Panic (trimmed):

```
panic: runtime error: invalid memory address or nil pointer dereference

helm.sh/helm/v4/pkg/kube.(*Client).IsReachable(...)
	pkg/kube/client.go:257
helm.sh/helm/v4/pkg/action.(*Install).RunWithContext(...)
	pkg/action/install.go:296
```

### What did you expect to happen?

The second call should try to create the client again and return the same "kubernetes cluster unreachable" error. It should not panic. A failed client creation should not be cached.


### How can we reproduce it (as minimally and precisely as possible)?

This is reproducible with the Go package only. No CLI and no kubectl are involved. The problem shows up whenever `IsReachable` is called twice on the same `kube.Client` after the first client creation failed:

```go
// Configure Helm to use a context that does not exist, then:
cfg := new(action.Configuration)
_ = cfg.Init(restClientGetterWithMissingContext, "default", "secret")

// First call returns a clean error.
err1 := cfg.KubeClient.IsReachable()
// err1 != nil, as expected

// Second call panics instead of returning the same error.
err2 := cfg.KubeClient.IsReachable()
// panic: nil pointer dereference
```

The same effect happens in real use when an action that checks reachability (such as `history`) runs before another action that checks reachability (such as `install`) on the same configuration, while the target context or kubeconfig cannot be resolved.

### Suggested fix

Only cache the client when creation succeeds, so a failure does not poison the cache:

```go
func (c *Client) getKubeClient() (kubernetes.Interface, error) {
	if c.kubeClient != nil {
		return c.kubeClient, nil
	}
	kc, err := c.Factory.KubernetesClientSet()
	if err != nil {
		return nil, err
	}
	c.kubeClient = kc
	return c.kubeClient, nil
}
```

An extra nil guard in `IsReachable` would also help, but fixing the cache is the real fix.

### Note

I would like to work on this and open a pull request with the fix and a regression test (calling `IsReachable` twice with a factory that returns an error on the first call). Please let me know if you are happy with this approach before I start.

### Helm version

<details>

Not using the Helm CLI. Helm is used as a Go library.
Module: helm.sh/helm/v4 v4.2.0 (the same code is also present on the main branch)
Go: go1.26.3

</details>


### Kubernetes version

<details>

Not using kubectl. The target is a local Kind cluster.
Note: the bug does not depend on the Kubernetes server version, because it happens
before any successful connection to the API server.

</details>

--- a/pkg/kube/client.go
+++ b/pkg/kube/client.go
@@ -235,12 +235,15 @@ func New(getter genericclioptions.RESTClientGetter) *Client {
 
 // getKubeClient get or create a new KubernetesClientSet
 func (c *Client) getKubeClient() (kubernetes.Interface, error) {
-	var err error
-	if c.kubeClient == nil {
-		c.kubeClient, err = c.Factory.KubernetesClientSet()
+	if c.kubeClient != nil {
+		return c.kubeClient, nil
 	}
-
-	return c.kubeClient, err
+	kc, err := c.Factory.KubernetesClientSet()
+	if err != nil {
+		return nil, err
+	}
+	c.kubeClient = kc
+	return c.kubeClient, nil
 }
 
 // IsReachable tests connectivity to the cluster.

--- a/pkg/kube/client_test.go
+++ b/pkg/kube/client_test.go
@@ -1416,6 +1416,26 @@ func TestIsReachable(t *testing.T) {
 	}
 }
 
+func TestIsReachableTwiceAfterClientCreationFailure(t *testing.T) {
+	refusedErr := errors.New("connection refused")
+	client := newTestClient(t)
+	client.Factory = &errorFactory{err: refusedErr}
+
+	assertReachableErr := func(label string, err error) {
+		t.Helper()
+		if err == nil {
+			t.Fatalf("%s: expected error, got nil", label)
+		}
+		if !errors.Is(err, refusedErr) {
+			t.Fatalf("%s: expected error wrapping %v, got %v", label, refusedErr, err)
+		}
+	}
+
+	assertReachableErr("first call", client.IsReachable())
+	// Second call must return the same underlying error, not panic.
+	assertReachableErr("second call", client.IsReachable())
+}
+
 func TestIsIncompatibleServerError(t *testing.T) {
 	testCases := map[string]struct {
 		Err  error

